### ISSUE TITLE
.item() on a DataArray with dtype='datetime64[ns]' returns int

### ISSUE THREAD

**Issue by IvoCrnkovic** [pydata/xarray#3256]
#### MCVE Code Sample

```python
import datetime
import xarray as xr

test_da = xr.DataArray(datetime.datetime(2019, 1, 1, 1, 1))

test_da.item()
```

#### Expected Output
1546304460000000000

#### Problem Description
I would think it would be nice to get a `datetime` out of the `.item()` call then the nanosecond representation.

#### Output of ``xr.show_versions()``
<details>
When I call xr.show_versions() i get an error but im running xarray 0.12.3
</details>

**Comment by shoyer** [comment]
This is basically a direct manifestation of this NumPy bug: https://github.com/numpy/numpy/issues/12550

We could potentially had a workaround in xarray but it would be good to figure out the NumPy fix first, so we can be consistent.

**Comment by dopplershift** [comment]
Well it'd be nice to have something, because what we have to do right now is:
```python
times[0].values.astype('datetime64[ms]').astype('O')
```
which is an abomination and terrible for teaching.

**Comment by jthielen** [comment]
Would it make sense to add something to the `dt` accessor, such `as_datetime`? @dopplershift's example would still be somewhat verbose, but it seems more sensible:

```python
times[0].dt.as_datetime().item()
```

**Comment by dopplershift** [comment]
Yeah, it's not much shorter, but it's a much easier concept for new users to grasp, IMO.
